engine/player/unique_gear_dragonflight.cpp

```diff
@@ -7440,6 +7440,91 @@ void dreambinder_loom_of_the_great_cycle( special_effect_t& effect )
   effect.execute_action = create_proc_action<dreambinder_loom_of_the_great_cycle_t>( "web_of_dreams", effect );
 }
 
+// Thorncaller Claw
+// 424406 Driver/Values
+// 424965 Thorn Spirit DoT
+// 425177 Thorn Burst ICD + Proc Flags
+// 425181 Thorn Burst Damage
+// TODO: Implement on-death effect
+void thorncaller_claw( special_effect_t& effect ) {
+  struct thorn_spirit_t : public generic_proc_t
+  {
+    thorn_spirit_t( const special_effect_t& effect )
+      : generic_proc_t( effect, "thorn_spirit", effect.player->find_spell( 424965 ) )
+    {
+      base_td = effect.driver()->effectN( 2 ).average( effect.item );
+    }
+
+    void impact( action_state_t* s ) override
+    {
+      generic_proc_t::impact( s );
+      player->get_target_data( s->target )->debuff.thorn_spirit->trigger();
+    }
+  };
+
+  // 2023-10-13: Split damage has generic AOE scaling, confirmed via testing
+  auto thorn_burst_damage = create_proc_action<generic_aoe_proc_t>( "thorn_burst", effect, "thorn_burst", effect.player->find_spell( 425181 ), true );
+  thorn_burst_damage->base_dd_min = thorn_burst_damage->base_dd_max = effect.driver()->effectN( 3 ).average( effect.item );
+
+  auto thorn_burst            = new special_effect_t( effect.player );
+  thorn_burst->name_str       = "thorn_burst";
+  thorn_burst->type           = SPECIAL_EFFECT_EQUIP;
+  thorn_burst->source         = SPECIAL_EFFECT_SOURCE_ITEM;
+  thorn_burst->spell_id       = 425177;
+  thorn_burst->execute_action = thorn_burst_damage;
+  effect.player->special_effects.push_back( thorn_burst );
+
+  auto thorn_burst_proc = new dbc_proc_callback_t( effect.player, *thorn_burst );
+  thorn_burst_proc->initialize();
+  thorn_burst_proc->activate();
+
+  effect.player->callbacks.register_callback_trigger_function(
+      425177, dbc_proc_callback_t::trigger_fn_type::CONDITION,
+      [ effect ]( const dbc_proc_callback_t*, action_t*, action_state_t* s ) {
+        auto td = effect.player->get_target_data( s->target )->debuff.thorn_spirit;
+        return td->check();
+      } );
+
+  auto thorn_spirit = create_proc_action<thorn_spirit_t>( "thorn_spirit", effect );
+  thorn_spirit->add_child( thorn_burst_damage );
+
+  // 2023-10-14: When Thorn Spirit spreads to a new target on demise it is applied with full duration.
+  // Can spread to a target with an existing Thorn Spirit even if other targets are in range, refreshes normally.
+  range::for_each( effect.player->sim->actor_list, [ effect, thorn_spirit ]( player_t* target ) {
+    target->register_on_demise_callback( effect.player, [ effect, thorn_spirit ]( player_t* t ) {
+      thorn_spirit->target_cache.is_valid = false;
+      std::vector<player_t*> targets      = thorn_spirit->target_list();
+      if ( targets.size() != 0 )
+      {
+        // Choose a random new target to spread to
+        player_t* new_target =
+            targets[ static_cast<int>( effect.player->rng().range( 0, static_cast<double>( targets.size() ) ) ) ];
```

Comment:
for simc specifically, we should use `debug_cast` in these situations, rather than a `static_cast`, while its converted to a `static_cast` during compilation with the release config, helps for debugging if something goes wrong.